packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1647,7 +1646,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         assert(summarizeResult.summary.type === SummaryType.Tree,
             0x12f /* "Container Runtime's summarize should always return a tree" */);
 
-        return summarizeResult as IChannelSummarizeResult;
+        return summarizeResult as ISummaryTreeWithStats;
```

Comment:
why do we need a cast here?

Reply:
as casting is always dangerous and should be avoided

Reply:
The compiler complains if I don't cast this.
In the layers below (data store context and channel context), we have a context and a channel. Context returns ISummarizeResult and channel returns ISummaryTreeWithStats. However, in container runtime, the same layer acts as both. Ideally, this would just return ISummaryTreeWithStats and a context above should have summarizeInternal and return ISummarizeResult.

Also note that there is an assert above this line which verifies that the summary returned is infact a tree and not a handle. So, this cast should be safe.

Reply:
can you just add a function that asserts the type, `funtion foo(obj:any): asserts obj is ISummaryTreeWithStats` that will ensure that the assert and the cast always happen together:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

Reply:
That's a good way to get around this. Thanks for the suggestion.